mentat/include_files.py

```diff
@@ -85,27 +72,29 @@ def abs_files_from_list(paths: list[Path], check_for_text: bool = True):
                 )
             )
 
-    files_from_dirs = [
-        CodeFile(os.path.realpath(path)) for path in file_paths_from_dirs
-    ]
-    return files_direct, files_from_dirs
+    files_from_dirs = set(CodeFile(path.resolve()) for path in file_paths_from_dirs)
+    return files_direct, files_from_dirs, invalid_paths
 
 
 def get_include_files(
     paths: list[Path], exclude_paths: list[Path]
-) -> Dict[Path, CodeFile]:
+) -> tuple[Dict[Path, CodeFile], list[str]]:
     """Returns a complete list of text files in a given set of include/exclude Paths."""
```

Comment:
I think we should keep the previous behavior of not returning the invalid paths, and just return a dictionary of `CodeFile` where all the code files with invalid paths are filtered out. We don't use the returned list of `invalid_paths` other than just sending them to the `SessionStream` (which can be done in this method) anyways.

Reply:
Nicer to just return 1 object and not a tuple of objects also imo

Reply:
+1

Reply:
Well if we send it to sessionstream in this method not only do we have to make this method (and a few others) async, but we have no control over if we want to print the invalid files or not; additionally, I don't want to give our utility functions random side-effects (like printing), because it makes things a lot less clear and easily leads to bugs (where you use a function not realizing it was hand-built for this one purpose and has a lot of random side-effects). I agree it's messy having to return the invalid_paths, but I can't think of a better way; let me know if y'all think of anything though!

Reply:
>we send it to sessionstream in this method not only do we have to make this method (and a few others) async

I removed the method to synchronously enqueue stream messages (pushes into a queue without `await`, we don't have any queue size limits anyways), but maybe we should re-add this if it makes the development experience better. Thinking about it now, I don't see any downsides to re-add. 

>we have no control over if we want to print the invalid files or not

we could add a `print_invalid_files: bool = True` argument that toggles the prints

> random side-effects (like printing)

I feel like prints aren't a random side effect? Similar to log calls, I think its totally normal to have them wherever in your code base since they're capturing relevant context to whatever function/method they're being called from.